### PR TITLE
Tie alt to manage object icon object class

### PR DESCRIPTION
**Pull request by jsoref**
The jelly is designed to give reasonable alt text. What's present now is actually worse than nothing.

```
<a href="configure" title="Configure System"><img src="/static/f12bd9f6/images/48x48/gear2.png" alt="manage-option" class="icon" /><dl><dt>Configure System</dt><dd>Configure global settings and paths.</dd><dd></dd></dl></a></div><div class="manage-option">
  |  
  | <a href="configureSecurity" title="Configure Global Security"><img src="/static/f12bd9f6/images/48x48/secure.png" alt="manage-option" class="icon" /><dl><dt>Configure Global Security</dt><dd>Secure Jenkins; define who is allowed to access/use the system.</dd><dd></dd></dl></a></div><div class="manage-option">
```

A screen reader will offer `manage-option Configure System` and `manage-option Configure Global Security`. A human will see `gear Configure System` and `lock Configure Global Security`. Having the screen reader read `manage-option` repeatedly is actually worse than not having it read anything at all. And since the links are reasonably described...

That said, it seemed worth trying to express the spirit of the icons.

### CODE REVIEW

**Comment by daniel-beck** [comment]
Since the file names are frequently useless I wonder whether we could just empty the `alt` and done.

**Comment by jsoref** [comment]
I'm happy to do that.

**Review by oleg-nenashev — APPROVED** [review]
I am fine with the current approach. Leaving text empty is fine as well IMHO.

CC @recena who is the author of the new design

**Review by daniel-beck — APPROVED** [review]

**Comment by daniel-beck** [comment]
Would prefer empty since it doesn't add anything, but either works.

File: core/src/main/resources/jenkins/model/Jenkins/manage.jelly [hunk 1]
```diff
@@ -51,6 +51,7 @@ THE SOFTWARE.
     <j:forEach var="m" items="${it.managementLinks}">
       <l:hasPermission permission="${m.requiredPermission}">
         <j:set var="icon" value="${m.iconClassName != null ? m.iconClassName : m.iconFileName}" />
```
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/manage.jelly]: `getIconFileName` is `@CheckForNull`, so the line below could through NPEs I think.

**Review by daniel-beck — CHANGES_REQUESTED** [review]

File: core/src/main/resources/jenkins/model/Jenkins/manage.jelly [hunk 2]
```diff
@@ -51,6 +51,7 @@ THE SOFTWARE.
...
+        <j:set var="alt" value="${icon.replaceAll('\\d*\\.[^.]+$', '')} />
         <j:if test="${icon!=null}">
```
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/manage.jelly]: Null test right here, so just needs to be in this block.

File: core/src/main/resources/jenkins/model/Jenkins/manage.jelly [hunk 3]
```diff
@@ -54,6 +54,7 @@ THE SOFTWARE.
...
           <div class="manage-option">
           <j:set var="iconUrl" value="${icon.startsWith('/') ? resURL+icon : imagesURL + '/48x48/' + icon}" />
+          <j:set var="alt" value="${icon.replaceAll('\\d*\\.[^.]+$', '')} />
```
Comment (jsoref) [inline core/src/main/resources/jenkins/model/Jenkins/manage.jelly]: @daniel-beck : sorry, code updated by moving within the null check...

**Review by recena — APPROVED** [review]

File: core/src/main/resources/jenkins/model/Jenkins/manage.jelly [hunk 4]
```diff
@@ -68,7 +69,7 @@ THE SOFTWARE.
             </j:when>
             <j:when test="${m.requiresPOST}">
               <f:link href="${m.urlName}" post="${m.requiresPOST}">
-                <img class="icon" src="${iconUrl}" alt="manage-option"/>
+                <img class="icon" src="${iconUrl}" alt="[${%alt}]"/>
```
Comment (recena) [inline core/src/main/resources/jenkins/model/Jenkins/manage.jelly]: Given to the size of the images is known, I would add the attributes `width` and `height`.  [Ref](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Images_in_HTML)

**Review by daniel-beck — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@daniel-beck @jsoref Closing quote was missing. I have added it and retriggered CI. Please do not merge it until it passes

**Comment by jsoref** [comment]
@oleg-nenashev: thanks

**Comment by daniel-beck** [comment]
Un-onholding this due to build passing.

**Pull request merged**